### ISSUE TITLE
Failed to Image::makeFromEncoded

### ISSUE THREAD

**Issue by DATL4G** [Kamel-Media/Kamel#42]
Loading some image URLs fail on Desktop JVM. (Not tested on Android)

Exception: ```Failed to Image::makeFromEncoded```

Example URL: https://img-cdn.hltv.org/teamlogo/6LVaK1MVX06kO1ChAq22v6.svg?ixlib=java-2.1.0&s=1ad0be7366aab3ef6825996762c3e615

My KamelConfig:

```kotlin
KamelConfig {
    takeFrom(KamelConfig.Default)
    resourcesFetcher()
    svgDecoder()
    imageVectorDecoder()
}
```

Example Image display:

```kotlin
KamelImage(
	resource = asyncPainterResource("https://img-cdn.hltv.org/teamlogo/6LVaK1MVX06kO1ChAq22v6.svg?ixlib=java-2.1.0&s=1ad0be7366aab3ef6825996762c3e615"),
	contentDescription = null
)
```

**Comment by luca992** [comment]
that's a svg. And the only way the library currently detects svgs is having a `.svg` extension. If you add `.svg` to the url: `https://img-cdn.hltv.org/teamlogo/6LVaK1MVX06kO1ChAq22v6.svg?ixlib=java-2.1.0&s=1ad0be7366aab3ef6825996762c3e615.url` it will work... kinda... I just tried and unfortunately there's an issue rendering it:

<img width="675" alt="Screenshot 2023-06-24 at 9 29 45 PM" src="https://github.com/Kamel-Media/Kamel/assets/4157455/5c41dc18-7d89-43db-8101-04ad8f2c3f7b">

**Comment by luca992** [comment]
svg's are not working on android yet fyi. So that definitely won't work

**Comment by DATL4G** [comment]
The URL already contains `.svg` as ending, so it should detect it.
I guess it doesn't because of the query parameter.

**Comment by luca992** [comment]
Yeah it would have to end in `.svg` not contain it atm.

**Comment by luca992** [comment]
This is the issue. 
```kotlin
        when (data.toString().substringAfterLast(".")) {
            "svg" -> kamelConfig.loadSvgResource(data, resourceConfig)
            "xml" -> kamelConfig.loadImageVectorResource(data, resourceConfig)
            else -> kamelConfig.loadImageBitmapResource(data, resourceConfig)
        }
```

**Comment by DATL4G** [comment]
Yeah I think it should parse this as a Uri or Url and just take the path and then check the ending.
As fallback (when it can't be parsed) the existing method would be fine I guess

**Comment by luca992** [comment]
released in `v0.6.0` Thanks again!
